src/test/java/edu/harvard/iq/dataverse/util/cache/CacheFactoryBeanTest.java

```diff
@@ -0,0 +1,360 @@
+                cache.rateLimitCache = new TestCache(getConfig());
+            }
+
+            // Clear the static data, so it can be reloaded with the new mocked data
+            RateLimitUtil.rateLimitMap.clear();
+            RateLimitUtil.rateLimits.clear();
+        }
+
+        // Reset to default auth user
+        authUser.setRateLimitTier(1);
+        authUser.setSuperuser(false);
+        authUser.setUserIdentifier("authUser");
+
+        // Create a unique action for each test
+        action = "cmd-" + UUID.randomUUID();
+    }
+
+    @AfterAll
+    public static void cleanup() {
+        Hazelcast.shutdownAll();
+    }
+    @Test
+    public void testGuestUserGettingRateLimited() {
+        boolean rateLimited = false;
+        int cnt = 0;
+        for (; cnt <100; cnt++) {
+            rateLimited = !cache.checkRate(guestUser, action);
+            if (rateLimited) {
+                break;
+            }
+        }
+        String key = RateLimitUtil.generateCacheKey(guestUser, action);
+        assertTrue(cache.rateLimitCache.containsKey(key));
+        assertTrue(rateLimited && cnt > 1 && cnt <= 30, "rateLimited:"+rateLimited + " cnt:"+cnt);
+    }
+
+    @Test
+    public void testAdminUserExemptFromGettingRateLimited() {
+        authUser.setSuperuser(true);
+        authUser.setUserIdentifier("admin");
+        boolean rateLimited = false;
+        int cnt = 0;
+        for (; cnt <100; cnt++) {
+            rateLimited = !cache.checkRate(authUser, action);
+            if (rateLimited) {
+                break;
+            }
+        }
+        assertTrue(!rateLimited && cnt >= 99, "rateLimited:"+rateLimited + " cnt:"+cnt);
+    }
+
+    @Test
+    public void testAuthenticatedUserGettingRateLimited() throws InterruptedException {
+        authUser.setRateLimitTier(2); // 120 cals per hour - 1 added token every 30 seconds
+        boolean rateLimited = false;
+        int cnt;
+        for (cnt = 0; cnt <200; cnt++) {
+            rateLimited = !cache.checkRate(authUser, action);
+            if (rateLimited) {
+                break;
+            }
+        }
+        assertTrue(rateLimited && cnt == 120, "rateLimited:"+rateLimited + " cnt:"+cnt);
+
+        for (cnt = 0; cnt <60; cnt++) {
+            Thread.sleep(1000);// Wait for bucket to be replenished (check each second for 1 minute max)
+            rateLimited = !cache.checkRate(authUser, action);
+            if (!rateLimited) {
+                break;
+            }
+        }
+        assertTrue(!rateLimited, "rateLimited:"+rateLimited + " cnt:"+cnt);
+
+        // Now change the user's tier, so it is no longer limited
+        authUser.setRateLimitTier(3); // tier 3 = no limit
+        for (cnt = 0; cnt <200; cnt++) {
+            rateLimited = !cache.checkRate(authUser, action);
+            if (rateLimited) {
+                break;
+            }
+        }
+        assertTrue(!rateLimited && cnt == 200, "rateLimited:"+rateLimited + " cnt:"+cnt);
```

Comment:
**chore**: This should be `assertFalse` and the `cnt` assertion should be on its own (`assertEqual`).

Reply:
split to use assertFalse and assertEquals